scripts/pylint_extensions_test.py

```diff
@@ -259,3 +259,36 @@ def test(test_var_one, test_var_two, self): #@
             ),
         ):
             checker_test_object.checker.visit_functiondef(functiondef_node2)
+
+
+class RestrictedImportCheckerTests(unittest.TestCase):
+
+    def test_detect_restricted_import(self):
+        checker_test_object = testutils.CheckerTestCase()
+        checker_test_object.CHECKER_CLASS = (
+            pylint_extensions.RestrictedImportChecker)
+        checker_test_object.setup_method()
+
+        node_import = astroid.extract_node("""
+            import core.domain.acl_decorators #@
+        """)
+        node_import.root().name = 'oppia.core.storage.topic'
+        with checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='invalid-import',
+                node=node_import,
+            ),
+        ):
+            checker_test_object.checker.visit_import(node_import)
+
+        node_importfrom = astroid.extract_node("""
+            from core.domain import acl_decorators #@
+        """)
+        node_importfrom.root().name = 'oppia.core.storage.topic'
+        with checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='invalid-import',
+                node=node_importfrom,
+            )
+        ):
+            checker_test_object.checker.visit_importfrom(node_importfrom)
```

Comment:
Can you also add two test cases which do not produce any error?

Reply:
Done!